src/crypto/qat/qat_crypto_accel.cc

```diff
@@ -40,3 +40,29 @@ bool QccCryptoAccel::cbc_decrypt(unsigned char* out, const unsigned char* in, si
       const_cast<unsigned char *>(&iv[0]),
       const_cast<unsigned char *>(&key[0]), CPA_CY_SYM_CIPHER_DIRECTION_DECRYPT);
 }
+
+bool QccCryptoAccel::cbc_encrypt_batch(unsigned char* out, const unsigned char* in, size_t chunk_size, size_t total_len,
+        const unsigned char iv[][AES_256_IVSIZE],
+        const unsigned char (&key)[AES_256_KEYSIZE]) {
+  if ((total_len % AES_256_IVSIZE) != 0) {
+    return false;
+  }
+
```

Comment:
I mean here that whether we can still offload if there is no alignment. We offload for the aligned part, and for the remaining with the software, then compose it. Will this be any issue?

Reply:
Hi @optimistyzy ,  the code is implemented in the file.
https://github.com/ceph/ceph/blob/edd4e553efd3934292c768b39d9ca1ff8d920ef1/src/rgw/rgw_crypt.cc#L478-L497
https://github.com/ceph/ceph/blob/edd4e553efd3934292c768b39d9ca1ff8d920ef1/src/rgw/rgw_crypt.cc#L533-L552